src/dpapi_ng/_client.py

```diff
@@ -509,6 +572,94 @@ def ncrypt_unprotect_secret(
     return _decrypt_blob(blob, rk)
 
 
+def ncrypt_protect_secret(
+    data: bytes,
+    sid: string,
```

Comment:
I would rename this to `protection_descriptor`. Right now we only support SIDs but this could change in the future to use the SDDL/Certificate types.

```suggestion
    protection_descriptor: str,
```